pkg/storage/streams/store.go

```diff
@@ -413,3 +423,14 @@ func (lr *lazySegmentRanger) Range(ctx context.Context, offset, length int64) (i
 	}
 	return lr.ranger.Range(ctx, offset, length)
 }
+
+// CancelHandler handles clean up of segments on receiving CTRL+C
+func (s *streamStore) CancelHandler(ctx context.Context, totalSegments int64, path paths.Path) {
+	for i := 0; i < int(totalSegments); i++ {
+		currentPath := fmt.Sprintf("s%d", i)
```

Comment:
Use the streamstore helper method `getSegmentPath` instead of `fmt.Sprintf` + `path.Prepend`

Reply:
`currentPath := getSegmentPath(path, i)`

Then Below,

`err := s.segments.Delete(ctx, currentPath)`

`zap.S().Warnf("......", currentPath, err)`

Reply:
done

Reply:
modified as suggested